src/app/static/src/app/store/adr/actions.ts

```diff
@@ -144,32 +144,49 @@ export const actions: ActionTree<ADRState, RootState> & ADRActions = {
 
-        for (let i = 0; i < uploadFilesPayload.length; i++) {
-            commit({type: ADRMutation.ADRUploadProgress, payload: i + 1});
-            const { resourceType, resourceFilename, resourceId } = uploadFilesPayload[i]
+        if (selectedDatasetId){
+            commit({type: ADRMutation.ADRUploadStarted, payload: uploadFilesPayload.length});
 
-            const requestParams: Dict<string> = {resourceFileName: resourceFilename}
-            if (resourceId){
-                requestParams["resourceId"] = resourceId
-            }
+            for (let i = 0; i < uploadFilesPayload.length; i++) {
+                commit({type: ADRMutation.ADRUploadProgress, payload: i + 1});
+                const { resourceType, resourceFilename, resourceId } = uploadFilesPayload[i]
 
-            let apiRequest = api<ADRMutation, ADRMutation>(context)
-                                        .withError(ADRMutation.SetADRUploadError);
-            if  (i === uploadFilesPayload.length - 1) {
-                apiRequest = apiRequest.withSuccess(ADRMutation.ADRUploadCompleted);
-            } else {
-                apiRequest = apiRequest.ignoreSuccess();
-            }
-            const response = await apiRequest.postAndReturn(`/adr/datasets/${selectedDatasetId}/resource/${resourceType}/${modelCalibrateId}`,
-                qs.stringify(requestParams))
-            if (!response) {
-                break
+                const requestParams: Dict<string> = {resourceFileName: resourceFilename}
+                if (resourceId){
+                    requestParams["resourceId"] = resourceId
+                }
+
+                let apiRequest = api<ADRMutation, ADRMutation>(context)
+                                            .withError(ADRMutation.SetADRUploadError);
+                if  (i === uploadFilesPayload.length - 1) {
+                    apiRequest = apiRequest.withSuccess(ADRMutation.ADRUploadCompleted);
+                } else {
+                    apiRequest = apiRequest.ignoreSuccess();
+                }
+                const response = await apiRequest.postAndReturn(`/adr/datasets/${selectedDatasetId}/resource/${resourceType}/${modelCalibrateId}`,
+                    qs.stringify(requestParams))
+                if (!response) {
+                    break
+                }
             }
+            let datasets = state.datasets;
+            await api(context)
+                .ignoreErrors()
+                .ignoreSuccess()
+                .get(`/adr/datasets/${selectedDatasetId}`)
+                .then((response) => {
+                    if (response) {
+                        datasets = [response.data];
+                    }
+                });
+    
+            const regenDataset = datasetFromMetadata(selectedDatasetId, datasets, state.schemas!);
+            commit(`baseline/${BaselineMutation.SetDataset}`, regenDataset, {root: true});
```

Comment:
Some obvious duplication with `getUserCanUpload()`... would it make sense to share any of this code? Could relate to next point too.

Reply:
Yeah, feels like this could at least be an async method, called from here and getUserCanUpload, which takes a selectedDatasetId, regenerates the dataset and commits the mutation. getUserCanUpload should then be able to read the org Id it needs from the newly mutated state.